File: app/schemas/documento_oficial.py
"""Pydantic schemas for DocumentoOficial entity."""

from datetime import date, datetime

from pydantic import BaseModel, Field

from app.models.enums import DocumentoEntidade, DocumentoStatus


class DocumentoOficialBase(BaseModel):
    entidade: DocumentoEntidade
    entidade_id: int
    tipo_documento: str = Field(..., max_length=120)
    descricao: str | None = Field(None, max_length=255)
    nome_arquivo: str = Field(..., max_length=255)
    content_type: str = Field(..., max_length=120)
    s3_key: str
    bucket_s3: str | None = Field(None, max_length=255)
    data_emissao: date | None = None
    data_validade: date | None = None


class DocumentoOficialCreate(DocumentoOficialBase):
    pass


class DocumentoOficialUpdate(BaseModel):
    descricao: str | None = Field(None, max_length=255)
    data_emissao: date | None = None
    data_validade: date | None = None
    status: DocumentoStatus | None = None


class DocumentoOficialResponse(DocumentoOficialBase):
    id: int
    status: DocumentoStatus
    enviado_por: int | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/documento-oficial.ts
/** TypeScript interfaces for DocumentoOficial entity matching Pydantic schemas. */

import type { DocumentoEntidade, DocumentoStatus } from './enums';

export interface DocumentoOficialResponse {
  id: number;
  entidade: DocumentoEntidade;
  entidade_id: number;
  tipo_documento: string;
  descricao: string | null;
  nome_arquivo: string;
  content_type: string;
  s3_key: string;
  bucket_s3: string | null;
  data_emissao: string | null;
  data_validade: string | null;
  status: DocumentoStatus;
  enviado_por: number | null;
  created_at: string;
  updated_at: string;
}

export interface DocumentoOficialCreate {
  entidade: DocumentoEntidade;
  entidade_id: number;
  tipo_documento: string;
  descricao?: string | null;
  nome_arquivo: string;
  content_type: string;
  s3_key: string;
  bucket_s3?: string | null;
  data_emissao?: string | null;
  data_validade?: string | null;
}

export interface DocumentoOficialUpdate {
  descricao?: string | null;
  data_emissao?: string | null;
  data_validade?: string | null;
  status?: DocumentoStatus | null;
}
